src/main/java/io/jenkins/plugins/models/JiraIssue.java

```diff
@@ -0,0 +1,47 @@
+package io.jenkins.plugins.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class JiraIssue {
+    @JsonProperty("key")
+    public String key;
```

Comment:
private + getters + all args constructor ?

Reply:
Is that a question or a suggestion? 

Remember I can write java but don't know java traditions. Is there anything actually wrong with this? should I add setters? I don't think I need getters for json to work.

The issue with all args constructor is it'll take like 10 arguments, which seems super gross to me. Its not like its a library that will be used by others, its purely an internal things.

I'll defer to people who have more java experience.

Reply:
https://github.com/mkarneim/pojobuilder#annotating-the-pojo maybe?

Reply:
a reasonable explanation here:
https://stackoverflow.com/a/7959871/4951015

it probably doesn't matter here, but in anywhere that the API matters you really shouldn't do it as it makes it very difficult to evolve the API, as consumer directly access the data of an object rather than you being allowed to seamlessly migrate them somewhere else, basically it breaks encapsulation

lombok (https://projectlombok.org/) is the most common method of hiding the boiler plate but generating it, 	

but it's probably fine to just leave it

Reply:
oh yea for a public api for sure. I was just lazy. IdeaJ can easily generate it. I just didn't like the huge constructor thing.